pkg/storaged/content-views.jsx

```diff
@@ -891,11 +891,15 @@
                     SelectOne("purpose", _("Purpose"),
                               {
                                   value: "block",
-                                  choices: purposes
+                                  choices: purposes,
+                                  validate: val => {
+                                      if (val == "vdo" && need_vdo_install && client.features.packagekit_details.reason == "immutable-os")
+                                          return cockpit.format(_("The $0 package needs to be added to the OS image."), vdo_package);
+                                  }
                               }),
                     Message(cockpit.format(_("The $0 package will be installed to create VDO devices."), vdo_package),
                             {
-                                visible: vals => vals.purpose === 'vdo' && need_vdo_install,
+                                visible: vals => vals.purpose === 'vdo' && need_vdo_install && client.features.packagekit_details.available,
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18240-20230127-135755-f796be5b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)